volatility/framework/plugins/windows/cachedump.py

```diff
@@ -0,0 +1,131 @@
+
+from volatility.framework import interfaces, renderers
+from volatility.framework.configuration import requirements
+from volatility.framework.renderers import format_hints
+from volatility.framework.layers import intel
+from volatility.plugins.windows.registry import hivelist
+from volatility.plugins.windows import hashdump, lsadump, poolscanner
+from Crypto.Hash import HMAC
+from Crypto.Cipher import ARC4, AES
+from struct import unpack
+
+
+class Cachedump(interfaces.plugins.PluginInterface):
+    """Dumps lsa secrets from memory"""
+    @classmethod
+    def get_requirements(cls):
+        return [requirements.TranslationLayerRequirement(name = 'primary',
+                                                     description = 'Memory layer for the kernel',
+                                                     architectures = ["Intel32", "Intel64"]),
+            requirements.SymbolTableRequirement(name = "nt_symbols",
+                                                description = "Windows kernel symbols"),
+            requirements.PluginRequirement(name = 'hivelist', plugin = hivelist.HiveList, version = (1, 0, 0))
+            ]
+    @classmethod
+    def get_nlkm(cls, sechive, lsakey, is_vista_or_later):
+        return lsadump.Lsadump.get_secret_by_name(sechive, 'NL$KM', lsakey, is_vista_or_later)
+    
+    @classmethod
+    def decrypt_hash(cls, edata, nlkm, ch, xp):
+        if xp:
+            hmac_md5 = HMAC.new(nlkm.encode('latin1'), ch)
+            rc4key = hmac_md5.digest()
+            rc4 = ARC4.new(rc4key)
+            data = rc4.encrypt(edata)
+        else:
+            # based on  Based on code from http://lab.mediaservice.net/code/cachedump.rb
+            aes = AES.new(nlkm.encode('latin1')[16:32], AES.MODE_CBC, ch)
+            data = ""
+            for i in range(0, len(edata), 16):
+                buf = edata[i : i + 16]
+                if len(buf) < 16:
+                    buf += (16 - len(buf)) * "\00"
+                data += aes.decrypt(buf)
+        return data
+    
+    @classmethod
+    def parse_cache_entry(cls, cache_data):
+        (uname_len, domain_len) = unpack("<HH", cache_data[:4])
+        if len(cache_data[60:62]) == 0:
+            return (uname_len, domain_len, 0, '', '')
+        (domain_name_len,) = unpack("<H", cache_data[60:62])
+        ch = cache_data[64:80]
+        enc_data = cache_data[96:]
+        return (uname_len, domain_len, domain_name_len, enc_data, ch)
+
+    @classmethod
+    def parse_decrypted_cache(cls, dec_data, uname_len,
+            domain_len, domain_name_len):
+        uname_off = 72
+        pad = 2 * ((uname_len / 2) % 2)
+        domain_off = int(uname_off + uname_len + pad)
```

Comment:
Stylistically, I'd prefer `offset` than `off`, but in this case it is clear enough that this doesn't need clarification.

In general though, we prefer clear variable names, over something that a developer has to type a couple of times and then never touch again.  Particularly when autocomplete on IDEs makes it much easier to use longer variable names.